File: Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDLanguage.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOLanguage
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from language"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( language as ClLanguage) as Boolean
             Dim StSql as String = "Select * from language"
            StSql &= " Where  language_id='" & language.dataLanguageId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        language.DataName = dt.Rows(0)("name").ToString()
        language.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( language as ClLanguage)  as Integer
            Dim stSql as String = "insert into Language (" _
                       & " language_id ,name ,last_update" _
                           & " )values (" _
            & "'" & language.dataLanguageId & "'" _
                     & ",'" & language.dataName & "'" _
                     & ",'" & language.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( language as ClLanguage) as Integer
            
            Dim stSql as String =  "delete from Language " _
             & " Where  language_id='" & language.dataLanguageId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(language as ClLanguage) as Integer
            Dim stSql as String =  "update Language SET " _
             & " name = '" & language.DataName & "'" _
                             & ",last_update = '" & language.DataLastUpdate & "'" _
             & " Where  language_id='" & language.dataLanguageId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Net/Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDLanguage.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOLanguage 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  language_id ,name ,last_update  from language";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClLanguage language)  
        {
            String stSql = "Select  language_id ,name ,last_update  from language";
            stSql += " Where  language_id='" + language.LanguageId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        language.Name = dt.Rows[0]["name"].ToString();
        language.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClLanguage language)  
        {
            String stSql = "insert into Language (" 
                       + " language_id ,name ,last_update" 
                           + " )values (" 
            + "'" + language.LanguageId + "'" 
                     + ",'" + language.Name + "'" 
                     + ",'" + language.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClLanguage language) {
            
            String stSql  =  "delete from Language " 
             + " Where  language_id='" + language.LanguageId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClLanguage language ) {
            String stSql  =  "update Language SET " 
             + " name = '" + language.Name + "'" 
                             + ",last_update = '" + language.LastUpdate + "'" 
             + " Where  language_id='" + language.LanguageId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}
